R/nullModelFastScore.R

```diff
@@ -206,8 +206,9 @@ setMethod(".calcScore",
                n.obs <- .countNonMissing(geno, MARGIN = 2)
 
                # allele frequency
-               freq <- .alleleFreq(gdsobj, geno, sample.index=sample.index,
-                                        male.diploid=male.diploid, genome.build=genome.build)
+               chr <- chromWithPAR(gdsobj, genome.build=genome.build)
+               sex <- validateSex(gdsobj)[sample.index]
+               freq <- .alleleFreq(geno, chr, sex, male.diploid=male.diploid)
 
```

Comment:
@smgogarten - a lot of the code just above and below here is repeated from the new `.testGenoSingleBlock` function (count non-missing samples; filter monomorphs; mean impute). is it worth the effort to make what's replicated its own sub function?

Reply:
The same duplicated code is in assocTestAggregate. I made an internal function that can be used in all three places.